src/Storages/MergeTree/MergeTreeIndexText.h

```diff
@@ -195,6 +195,36 @@ struct DictionaryBlock : public DictionaryBlockBase
     std::vector<TokenPostingsInfo> token_infos;
 };
 
+class TextIndexHeader
+{
+public:
+    TextIndexHeader(size_t num_tokens_, BloomFilter bloom_filter_, DictionarySparseIndex sparse_index_)
+        : num_tokens(num_tokens_)
+        , bloom_filter(std::move(bloom_filter_))
+        , sparse_index(std::move(sparse_index_))
+    {
+    }
+
+    size_t numberOfTokens() const { return num_tokens; }
+    const BloomFilter & bloomFilter() const { return bloom_filter; }
+    const DictionarySparseIndex & sparseIndex() const { return sparse_index; }
+
+    size_t memoryUsageBytes() const
+    {
+        return sizeof(*this)
+            + bloom_filter.getFilterSizeBytes()
+            + sparse_index.tokens->allocatedBytes()
+            + sparse_index.offsets_in_file->allocatedBytes();
+    }
+
+private:
+    size_t num_tokens;
+    BloomFilter bloom_filter;
+    DictionarySparseIndex sparse_index;
+};
```

Comment:
Very minor: Classes like this I prefer to declare as nested classes, so the names get shorter and their purpose is more clear. But that's a personal preference.